flink-connectors/flink-connector-files/src/main/java/org/apache/flink/connector/file/src/reader/FileRecordFormat.java

```diff
@@ -147,7 +147,7 @@
      *
      * <p>The number should be large enough so that the thread-to-thread handover overhead is
      * amortized across the records, but small enough so that the these records together do not
-     * consume too memory to be feasible.
+     * consume too much memory to be feasible.
```

Comment:
I don't quite understand what feasible means in this context.

```suggestion
     * consume too much memory.
```

Can we give guidelines instead? Like increase the number until no more performance gain is observable?

Reply:
I think the original idea was to suggest using feasible memory, not too little, not to much.

Reply:
The question is if the user knows what is feasible? I find it rather fuzzy. The easiest option is to simply get rid of `to be feasible`.

Reply:
Maybe let me restructure the text to make it a little bit clear:

The number should be set to be feasible considering the thread-to-thread handover overhead for each fetch and the memory capacity of the source worker. On one hand, The number should be large enough so that the thread-to-thread handover overhead is amortized across the records, which means the overhead impact for each record should be acceptable. On the other hand, the number should be small enough so that the these records together do not consume too much memory, which depends on the real memory capacity of the source worker.

Reply:
It is fine to me to just get rid of "to be feasible".

Reply:
Daisy proposed `but small enough so that the records together provide the most optimal performance gains.`
It reads much better but is still a bit fuzzy I guess.

Reply:
What about a pattern like the following for more clarity?

"If the number of X is too high, then there won't be enough Y in each Z, which means that it will take longer (or use more memory) for ... "

"Conversely, if the number is too low then ..."

Reply:
I think the config of the number leads to find the right trade-off between t2t handover overhead and memory consumption. Both of them are related to the performance. If we only talk about performance gains, we are just talking about the end result (the "what" part) and missing the details information (the "why" part, i.e. lower overhead and feasible memory consumption mean better performance gains). Speaking of feasible memory consumption, it really depends on use cases, e.g. how big each record is, how much memory the work has, etc. We might be able to define the guideline, like each fetch should consume bigger than X but smaller than Y MB memory. But, the real number of X, Y should come from measurement of real applications/benchmark. I think this is out of the scope of javadoc and should described as best practices with online document or blog.

Reply:
> What about a pattern like the following for more clarity?
> 
> "If the number of X is too high, then there won't be enough Y in each Z, which means that it will take longer (or use more memory) for ... "
> 
> "Conversely, if the number is too low then ..."

yeap, this is the normal description of "small&large", which is (for me) better to understand than the opposite description of "small enough&large enough".